src/contentmanagerdelegate.cpp

```diff
@@ -285,10 +285,26 @@ void ContentManagerDelegate::handleLastColumnClicked(const QModelIndex& index, Q
 
 QSize ContentManagerDelegate::sizeHint(const QStyleOptionViewItem &option, const QModelIndex &index) const
 {
-    Q_UNUSED(option);
-
-    if (index.parent().isValid()) {
-        return QSize(300, 70);
+    if (index.parent().isValid())
+    {
+        auto view = KiwixApp::instance()->getContentManager()->getView()->getView();
+
+        /* Get the Rectangle from start of column1 to end of column 5. */
+        QRect r = view->visualRect(index.parent().siblingAtColumn(1));
+        QRect lastColRect = view->visualRect(index.parent().siblingAtColumn(5));
+        r.setWidth(lastColRect.x() + lastColRect.width() - r.x());
+
+        /* Based on the rectangle and text, find the best fitting size. */
+        QFontMetrics fm(option.font);
+        QSize textRectSize = fm
+            .boundingRect(r,
+                          Qt::AlignLeft | Qt::AlignVCenter | Qt::TextWordWrap,
+                          index.data().toString())
+            .size();
+        
+        if (textRectSize.height() < 70)
+            textRectSize.setHeight(70);
+        return textRectSize;
     }
```

Comment:
Nice, but it seems to slightly fall short of its goal:

![Screenshot from 2024-06-06 13-49-50](https://github.com/kiwix/kiwix-desktop/assets/11933502/988fcc75-dbbc-4251-a97e-11e753f96d1f)